packages/strapi-admin/admin/src/containers/Webhooks/EditView/tests/reducer.test.js

```diff
@@ -0,0 +1,265 @@
+        .push(fromJS(header));
+
+      const expectedState = state.setIn(
+        ['modifiedData', ...action.keys],
+        updatedHeaders
+      );
+
+      expect(reducer(state, action)).toEqual(expectedState);
+    });
+
+    it('should remove header from modifiedData if it is not the last', () => {
+      const initialHeaders = [
+        { key: 'accept', value: 'text/html' },
+        { key: 'authorization', value: 'Basic YWxhZGRpbjpvcGVuc2VzYW1l' },
+      ];
+      const state = initialState
+        .setIn(['initialData', 'headers'], fromJS(initialHeaders))
+        .setIn(['modifiedData', 'headers'], fromJS(initialHeaders));
+
+      const action = {
+        type: 'ON_HEADER_REMOVE',
+        index: 1,
+      };
+
+      const updatedHeaders = [{ key: 'accept', value: 'text/html' }];
+
+      const expectedState = state.setIn(
+        ['modifiedData', 'headers'],
+        fromJS(updatedHeaders)
+      );
+
+      expect(reducer(state, action)).toEqual(expectedState);
+    });
+
+    it('should clear a header to modifiedData if it is the last', () => {
+      const initialHeaders = [{ key: 'accept', value: 'text/html' }];
+      const state = initialState
+        .setIn(['initialData', 'headers'], fromJS(initialHeaders))
+        .setIn(['modifiedData', 'headers'], fromJS(initialHeaders));
+
+      const action = {
+        type: 'ON_HEADER_REMOVE',
+        index: 0,
+      };
+
+      const updatedHeaders = [header];
+
+      const expectedState = state.setIn(
+        ['modifiedData', 'headers'],
+        fromJS(updatedHeaders)
+      );
+
+      expect(reducer(state, action)).toEqual(expectedState);
+    });
+  });
+
+  describe('Trigger actions', () => {
+    it('should set isTriggering to false when trigger action is canceled', () => {
+      const state = initialState.set('isTriggering', true);
+
+      const action = {
+        type: 'ON_TRIGGER_CANCELED',
+      };
+
+      const expectedState = state.update('isTriggering', () => false);
+
+      expect(reducer(state, action)).toEqual(expectedState);
+    });
+
+    it('should clear triggerResponse when trigger action is canceled', () => {
+      const state = initialState.set('triggerResponse', {
+        statusCode: 200,
+        message: 'succeed',
+      });
+
+      const action = {
+        type: 'ON_TRIGGER_CANCELED',
+      };
+
+      const expectedState = state.update('triggerResponse', () => fromJS({}));
+
+      expect(reducer(state, action)).toEqual(expectedState);
+    });
+
+    it('should toggle isTriggering value', () => {
+      const state = initialState;
+
+      const action = {
+        type: 'SET_IS_TRIGGERING',
+      };
+
+      const expectedState = state.update(
+        'isTriggering',
+        isTriggering => !isTriggering
+      );
+
+      expect(reducer(state, action)).toEqual(expectedState);
+    });
+
+    it('should update isTriggering and triggerResponse if trigger succeed', () => {
+      const state = initialState;
```

Comment:
You may need to set the `isTriggering` key to `true` here otherwise you cannot test that the key is going from `true` to `false`